build_exercise_proj/tools_lib_proj/tools.hpp

```diff
@@ -1,2 +1,4 @@
-#pragma 
-void Greeting();
\ No newline at end of file
+#pragma  
```

Comment:
I would strongly advice against using #pragma.
This is a Visual Studio abomination, that is not standard C++.

Using header guards is the prefered option.
```
#ifndef TOOLS_LIB_PROJ_TOOLS_HPP_
#define TOOLS_LIB_PROJ_TOOLS_HPP_

// place code here

#endif //TOOLS_LIB_PROJ_TOOLS_HPP_
```

Reply:
At #endif is //TOOLS_LIB_PROJ_TOOLS_HPP_ a comment only for clarity which endif ends?